src/sntp.rs

```diff
@@ -113,6 +117,8 @@ impl<'a> Default for SntpConf<'a> {
     }
 }
 
+type SyncCallback = Box<dyn FnMut(Duration) + Send + 'static>;
```

Comment:
... and while we are on the `no_std` topic, please:
* declare `#[cfg(feature = "alloc")] extern crate alloc;`
* Protect with `#[cfg(feature = "alloc")]` your `SyncCallback` (note that you should turn `Box` into `alloc::boxed::Box`)
* Ditto for `SYNC_CB`